modules/world/evaluation/labels/safe_distance_label_function.cpp

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+#include "safe_distance_label_function.hpp"
+#include "modules/world/observed_world.hpp"
+
+namespace modules {
+namespace world {
+namespace evaluation {
+
+using modules::models::dynamic::StateDefinition;
+
+SafeDistanceLabelFunction::SafeDistanceLabelFunction(
+    const std::string& label_str, bool to_rear, double delta, double a_e,
+    double a_o)
+    : BaseLabelFunction(label_str),
+      to_rear_(to_rear),
+      delta_(delta),
+      a_e_(a_e),
+      a_o_(a_o) {}
+
+std::vector<LabelMap::value_type> SafeDistanceLabelFunction::Evaluate(
+    const world::ObservedWorld& observed_world) const {
+  auto ego = std::const_pointer_cast<Agent>(observed_world.GetEgoAgent());
+  auto lane_corridor =
+      ego->GetRoadCorridor()->GetNearestLaneCorridor(ego->GetCurrentPosition());
```

Comment:
why can't we use GetCurrentLaneCorridor() here?

Reply:
See https://github.com/bark-simulator/bark/pull/377#discussion_r442195201